### PR TITLE
feat(scale-test): scripts that generates, run apps and collect results

### PR DESCRIPTION
**Pull request by vishnuitta**
This `scale-test` directory contains scripts and templated yamls to generate required number of application pods yaml and apply these yamls.
After test is completed, scripts takes care of collecting fio outputs into local machine's <curdir>/<testid> and prints the IOPS/latency snippets of fio output.
These scripts also takes care of any cleanup required during the start of test, and also does cleanup at the end of test.

More info is updated in README.md

Signed-off-by: Vitta <vitta@mayadata.io>

### CODE REVIEW

**Comment by ksatchit** [comment]
Thanks for the scripts @vishnuitta !

File: scale-test/fioload-template.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    name: fio
+    volumeMounts:
+    - mountPath: /fio-results
+      name: results-volume
+    - mountPath: /mnt
+      name: data-volume
+  volumes:
+  - name: results-volume
+    hostPath:
+      path: /home/vitta/fio-results/@testid@
```
Comment (kmova) [inline scale-test/fioload-template.yaml]: n_ : instead of home/vitta, can a directory like /tmp/fio-results be provided.

**Review by kmova — APPROVED** [review]
I glanced through the changes once and looks good to me. An additional detailed review will be helpful.

File: scale-test/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+This `scale-test` directory contains scripts and templated yamls to generate required number of application pods yaml and apply these yamls.
```
Comment (gila) [inline scale-test/README.md:1]: Please remove the swap file from your editor of this commit

File: scale-test/fioload-template.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    app: @testid@
+spec:
+  restartPolicy: Never
+  @nodeSelector@:
+    type: worker
+  containers:
+  - command:
+    - sh
+    - -c
+    - 'fio --blocksize=@bs@ --verify=0 --randrepeat=0 --filename=/mnt/file1 --ioengine=libaio --readwrite=randread --runtime=60s --time_based --ramp_time=2s --size=3G --name=test --numjobs=8 --direct=1 --iodepth=2 --group_reporting > /tmp/fiotest; exit 0;'
```
Comment (gila) [inline scale-test/fioload-template.yaml:16]: Fio has an option to specify the output file, I would use that as it is also not needed to collect the interval updates that fio produces.

File: scale-test/fioload-template.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - command: ["/bin/sh", "-c"]
+    args:
+      - cnt=0;
+        while [ $cnt -ne 1 ];
+        do
+          sleep 5;
+          cnt=`ls -ltr /mnt/file1 2>/dev/null | wc -l`;
+        done;
+        sleep 5;
+        fio --blocksize=@bs@ --verify=0 --randrepeat=0 --filename=/mnt/file2 --ioengine=libaio --readwrite=randread --runtime=15s --time_based --ramp_time=2s --size=3G --name=test --direct=1 --group_reporting > /fio-results/@output@;
```
Comment (gila) [inline scale-test/fioload-template.yaml:31]: You are running fio twice? Can you explain what the idea behind that is? Also, why do you need to sleep for the file to be created? If you want to have jobs completed, or started in a certain order, the fio configuration files let you do that. 

Also, in both cases the verify option has no effect, you can remove it. Lastly the options used in the first test are not the same as in the last test.

File: scale-test/fiopvctemplate.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+apiVersion: v1
+metadata:
+  name: @pvc@
+spec:
+  storageClassName: cstor2
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10G
```
Comment (gila) [inline scale-test/fiopvctemplate.yaml:12]: You allocate 10GB but you only create a file of 3GB why is that?

File: scale-test/fioload-template.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (gila) [inline scale-test/fioload-template.yaml:1]: This only ever does read. By default, cstor is configured with 4GB iirc, so let's say its 3GB of ARC. Most IO will be in memory once the file is laid out. 

Also, the iodepth of 2 is not a realistic value, it should match the value that the target supports.

File: scale-test/fiopvctemplate.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: @pvc@
+spec:
+  storageClassName: cstor2
```
Comment (gila) [inline scale-test/fiopvctemplate.yaml:7]: where is the storage class defined?

File: scale-test/fiotemplate.yaml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    app: @testid@
+spec:
+  restartPolicy: Never
+  @nodeSelector@:
+    type: worker
+  containers:
+  - command:
+    - sh
+    - -c
+    - 'fio --blocksize=@bs@ --verify=0 --randrepeat=0 --filename=/mnt/file1 --ioengine=libaio --readwrite=randread --runtime=15s --time_based --ramp_time=2s --size=3G --name=test --numjobs=8 --direct=1 --iodepth=2 --group_reporting > /fio-results/@output@; exit 0;'
```
Comment (gila) [inline scale-test/fiotemplate.yaml:16]: For this, the same comments apply as to the other template file

**Review by gila — CHANGES_REQUESTED** [review]

**Pull request merged**